### PR TITLE
bgpv2: container labs for various types of advertisements

### PR DESCRIPTION
**Pull request by harsimran-pabla**
Adding container labs for local development and testing of BGPv2 APIs. 

PR is split by BGP advertisement type
- PodCIDR
- Multi-homing
- CiliumPodIPPool
- Service types : Loadbalancer, cluster IP, external IP

### CODE REVIEW

File: contrib/containerlab/bgpv2/pod-cidr/Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+deploy:
+	kind create cluster --config cluster.yaml
+	sudo containerlab -t topo.yaml deploy
+	# create secret for bgp
+	kubectl -n kube-system create secret generic --type=string bgp-auth-secret --from-literal=password=cilium123
+	# install cilium
+	helm install cilium -n kube-system oci://quay.io/cilium-charts-dev/cilium --version 1.16.0-dev-dev.247-main-8c97a2119d -f values.yaml
```
Comment (YutaroHayakawa) [inline contrib/containerlab/bgpv2/pod-cidr/Makefile]: I think we should be able to override this Cilium version.

**Review by YutaroHayakawa — CHANGES_REQUESTED** [review]
Nice work! I think we can merge PodCIDR and Service labs into the single lab as their ContainerLab and Cilium configurations are the same.

File: contrib/containerlab/bgpv2/multi-homing/bgp.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+metadata:
+  name: pod-cidr-advert
+  labels:
+    advertise: pod-cidr
+spec:
+  advertisements:
+    - advertisementType: "PodCIDR"
+      attributes:
+        communities:
+          wellKnown: [ "no-export" ]
```
Comment (bimmlerd) [inline contrib/containerlab/bgpv2/multi-homing/bgp.yaml]: nit: no newline at EOF

File: contrib/containerlab/bgpv2/multi-homing/Makefile [hunk 3]
```diff
@@ -0,0 +1,20 @@
+deploy:
+	kind create cluster --config cluster.yaml
+	sudo containerlab -t topo.yaml deploy
+	# create secret for bgp
+	kubectl -n kube-system create secret generic --type=string bgp-auth-secret --from-literal=password=cilium123
+	# install cilium
+	helm install cilium -n kube-system oci://quay.io/cilium-charts-dev/cilium --version 1.16.0-dev-dev.247-main-8c97a2119d -f values.yaml
```
Comment (bimmlerd) [inline contrib/containerlab/bgpv2/multi-homing/Makefile]: similarly here, this is a very specific cilium version

File: contrib/containerlab/bgpv2/multi-homing/Makefile [hunk 4]
```diff
@@ -0,0 +1,20 @@
+deploy:
+	kind create cluster --config cluster.yaml
+	sudo containerlab -t topo.yaml deploy
```
Comment (bimmlerd) [inline contrib/containerlab/bgpv2/multi-homing/Makefile:7]: could add a check for containerlab to print a more helpful message from where to install it if one doesn't yet have it

**Review by bimmlerd — CHANGES_REQUESTED** [review]
I'm a bit lost as to what these are - could we add a bit of documentation somewhere?

File: contrib/containerlab/bgpv2/pod-cidr/Makefile (hunk 1 above)
Comment (harsimran-pabla) [reply contrib/containerlab/bgpv2/pod-cidr/Makefile]: Added an option to override cilium version. This is temporary, till official release is done. At that point, I would just remove specific cilium version installation and get latest.

File: contrib/containerlab/bgpv2/multi-homing/Makefile (hunk 3 above)
Comment (harsimran-pabla) [reply contrib/containerlab/bgpv2/multi-homing/Makefile]: Made this configurable, waiting for next official release at which point I will remove it and get latest.

File: contrib/containerlab/bgpv2/multi-homing/Makefile (hunk 4 above)
Comment (harsimran-pabla) [reply contrib/containerlab/bgpv2/multi-homing/Makefile:7]: I wanted to keep these make files simple. I added a readme section under containerlab/bgpv2, which provides prerequisites of installing kind, containterlab etc. Hope that is enough for users to know what kind of binaries are required to run the lab. I think target audience probably already have container lab - if not, readme should help.

**Comment by harsimran-pabla** [comment]
Hi @bimmlerd 

> I'm a bit lost as to what these are - could we add a bit of documentation somewhere?

Sorry, I should have added readme section from the start. I added a new commit which contains description about each lab and prerequisite to get this working. https://github.com/cilium/cilium/pull/32522/commits/6e1285fea5471befd346e6519a308d6b08d4641e

File: contrib/containerlab/bgpv2/multi-homing/Makefile (hunk 3 above)
Comment (YutaroHayakawa) [reply contrib/containerlab/bgpv2/multi-homing/Makefile]: How did you make it configurable? I don't see the knob. Am I missing something?

Comment (harsimran-pabla) [reply contrib/containerlab/bgpv2/multi-homing/Makefile]: woops, commit got lost somewhere  :( checking

Comment (harsimran-pabla) [reply contrib/containerlab/bgpv2/multi-homing/Makefile]: Pushed the change again to have it fixed.

**Review by YutaroHayakawa — APPROVED** [review]

**Review by bimmlerd — APPROVED** [review]

**Comment by harsimran-pabla** [comment]
Not sure if a full test run is required for this change? There is no modification to the core Cilium code base, just new labs in contrib directory. @YutaroHayakawa  @joestringer any concerns about adding ready-to-merge label manually, since basic pull request checks have passed?

**Comment by joestringer** [comment]
Making an exception since this is just going into `contrib/`.

**Pull request merged**